main.py

```diff
@@ -0,0 +1,69 @@
+"""Proxy server to communicate a client to ARTIQ."""
+
+from contextlib import asynccontextmanager
+import json
+import os
+
+from fastapi import FastAPI
+from sipyco.pc_rpc import Client
+
+
+settings = {}
+
+
+def load_setup_file():
+    """Loads set-up information from the setup file.
+
+    The file should have the following JSON structure:
+
+      {
+        "master_path": {master_path}
+      }
+    """
+    global settings
```

Comment:
As you said, I don't like to use the `global` keyword either.
One workaround would be to pass it as an argument of the function.
Since dictionary is a mutable container, we can [`update()`](https://docs.python.org/3/library/stdtypes.html#dict.update) the contents in the function.

Reply:
Oh, it's a smart solution! I changed it using `update()` and it still runs correctly.